azavea/sentinel-2-stac-utils
# Add CLI skeleton

## Overview

This PR sets up the cli skeleton. It makes it so that there's a runnable command called `s2stac` that you can invoke with bloop that tells you about your options. Is it useful? No! But it will be someday.

### Checklist

- ~New tests have been added or existing tests have been modified~ nope!

### Notes

I borrowed our coercible implicit translator logic from PFP to get `Argument` evidence for newtypes, and man, [what a fun time](https://github.com/azavea/sentinel-2-stac-utils/pull/6/files#diff-f538b1da91364a120f90d15ae7f76a13R6-R8).

### Testing Instructions

- `./sbt bloopInstall`
- `bloop run s2stac -- --collection L1c --inventory-path bar --output-catalog-root 'oh no'`
- `bloop run s2stac -- --collection lol --inventory-path '' --output-catalog-root ''` -- should fail helpfully
- `bloop run s2stac -- --collection lol --inventory-path ''` -- should also fail helpfully

Closes #1
